deploy/helm/sumologic/templates/events/otelcol/statefulset.yaml

```diff
@@ -0,0 +1,166 @@
+{{- if eq (include "events.otelcol.enabled" .) "true" }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "sumologic.metadata.name.events.statefulset" . }}
+  labels:
+    app: {{ template "sumologic.labels.app.events.statefulset" . }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "sumologic.labels.app.events.pod" . }}
+  serviceName: {{ template "sumologic.metadata.name.events.service-headless" . }}
+  podManagementPolicy: "Parallel"
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/events/otelcol/configmap.yaml") . | sha256sum }}
+{{- if .Values.sumologic.podAnnotations }}
+{{ toYaml .Values.sumologic.podAnnotations | indent 8 }}
+{{- end }}
+{{- if .Values.otelevents.statefulset.podAnnotations }}
+{{ toYaml .Values.otelevents.statefulset.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "sumologic.labels.app.events.pod" . }}
+        {{- include "sumologic.labels.common" . | nindent 8 }}
+{{- if .Values.sumologic.podLabels }}
+{{ toYaml .Values.sumologic.podLabels | indent 8 }}
+{{- end }}
+{{- if .Values.otelevents.statefulset.podLabels }}
+{{ toYaml .Values.otelevents.statefulset.podLabels | indent 8 }}
+{{- end }}
```

Comment:
It would be nice to test this configurations as well in helm tests

Reply:
Sure, though I think we don't test this for any OT statefulset 😅

Reply:
We should, as this ends with fix PR afterwards :dagger: